src/stream/src/executor/sort_buffer.rs

```diff
@@ -58,12 +57,29 @@ pub struct SortBuffer<S: StateStore> {
 
     chunk_size: usize,
 
-    last_watermark: Option<ScalarImpl>,
+    last_output: Option<ScalarImpl>,
+
+    watermark: Option<ScalarImpl>,
 
     buffer: BTreeMap<SortBufferKey, SortBufferValue>,
 }
 
 impl<S: StateStore> SortBuffer<S> {
+    /// Update the watermark value.
+    pub fn set_watermark(&mut self, watermark: ScalarImpl) {
+        assert!(
+            {
+                if let Some(ref last_watermark) = self.watermark {
+                    &watermark >= last_watermark
+                } else {
```

Comment:
will we have functions such as `cmp_watermark` when we remove the `Ord` implementation on ScalarImpl?

Reply:
TBD later